packages/block-editor/src/hooks/child-layout.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+	__experimentalUnitControl as UnitControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useSetting from '../components/use-setting';
+
+function helpText( selfStretch ) {
+	switch ( selfStretch ) {
+		case 'fill':
+			return __( 'Stretch to fill available space.' );
+		case 'fixed':
+			return __( 'Specify a fixed width.' );
+		default:
+			return __( 'Fit contents.' );
+	}
+}
+
+/**
+ * Inspector control panel containing the child layout related configuration
+ *
+ * @param {Object} props                        Block props.
+ * @param {Object} props.attributes             Block attributes.
+ * @param {Object} props.setAttributes          Function to set block attributes.
+ * @param {Object} props.__unstableParentLayout
+ *
+ * @return {WPElement} child layout edit element.
+ */
+export function ChildLayoutEdit( {
+	attributes,
+	setAttributes,
+	__unstableParentLayout: parentLayout,
+} ) {
+	const { style = {} } = attributes;
+	const { layout: childLayout = {} } = style;
+	const { selfStretch, flexSize } = childLayout;
+
+	return (
+		<>
+			<ToggleGroupControl
+				label={ childLayoutOrientation( parentLayout ) }
+				value={ selfStretch || 'fit' }
+				help={ helpText( selfStretch ) }
+				onChange={ ( value ) => {
+					setAttributes( {
+						style: {
+							...style,
+							layout: {
+								...childLayout,
+								selfStretch: value,
+							},
+						},
+					} );
+				} }
+				isBlock={ true }
+			>
+				<ToggleGroupControlOption
+					key={ 'fit' }
+					value={ 'fit' }
+					label={ __( 'Fit' ) }
+				/>
+				<ToggleGroupControlOption
+					key={ 'fill' }
+					value={ 'fill' }
+					label={ __( 'Fill' ) }
+				/>
+				<ToggleGroupControlOption
+					key={ 'fixed' }
+					value={ 'fixed' }
+					label={ __( 'Fixed' ) }
+				/>
+			</ToggleGroupControl>
+			{ selfStretch === 'fixed' && (
+				<UnitControl
+					style={ { height: 'auto' } }
```

Comment:
Can we avoid the inline style at all? There were some recent efforts to avoid these on UnitControls and controls in general.

Reply:
I'm afraid we can't remove it until #45877 is fixed, otherwise the input blows out to full parent height:

<img width="269" alt="Screenshot 2022-11-22 at 10 32 20 am" src="https://user-images.githubusercontent.com/8096000/203179168-0882a8a0-8eda-4f64-8666-d20a45ce32e3.png">

`size={ '__unstable-large' }` doesn't work on UnitControl either.

Reply:
> I'm afraid we can't remove it until https://github.com/WordPress/gutenberg/issues/45877 is fixed, otherwise the input blows out to full parent height:

In previous instances where the style was still required, I have been requested to use a CSS class instead. 

> size={ '__unstable-large' } doesn't work on UnitControl either.

Are you only referring to the UnitControl here? 

The `size` prop with `__unstable-large` is already in use and appears to work fine for the following components' UnitControls:
- border radius control
- spacing sizes control
- min-height
- and several others that pass through the size prop such as BorderBoxControl, FontSizePicker etc

Reply:
This UnitControl. I've gone with @mirka 's [recommendation](https://github.com/WordPress/gutenberg/pull/45364#discussion_r1026084154) for now; anything else should be fine to address as a follow-up.

Reply:
Thanks for the clarification 👍  

These controls still need to be 40px tall (as per this [confirmation](https://github.com/WordPress/gutenberg/pull/45364#discussion_r1027784985)) before landing this though, so they are consistent with the other controls in the panel.

Reply:
This works for me regarding fixing the control heights:

<details>
<summary>Example diff</summary>

```diff
diff --git a/packages/block-editor/src/hooks/child-layout.js b/packages/block-editor/src/hooks/child-layout.js
index 6ae08b221f..e29e1d9ac0 100644
--- a/packages/block-editor/src/hooks/child-layout.js
+++ b/packages/block-editor/src/hooks/child-layout.js
@@ -63,6 +63,7 @@ export function ChildLayoutEdit( {
 					} );
 				} }
 				isBlock={ true }
+				size={ '__unstable-large' }
 			>
 				<ToggleGroupControlOption
 					key={ 'fit' }
@@ -82,6 +83,7 @@ export function ChildLayoutEdit( {
 			</ToggleGroupControl>
 			{ selfStretch === 'fixed' && (
 				<UnitControl
+					size={ '__unstable-large' }
 					style={ { height: 'auto' } }
 					onChange={ ( value ) => {
 						setAttributes( {

```
</details>

Reply:
Oh right, I was trying it instead of the auto height; it does seem to work in addition to it. Fixing now, thanks!